crates/lair_keystore_api/src/internal/wire.rs

```diff
@@ -494,6 +502,303 @@ macro_rules! wire_type_meta_macro {
                     signature: signature.into(),
                 }
             },
+            ToLairX25519NewFromEntropy 0x00000242 false true {
+            } |msg_id, wire_type| {
+                let writer = default_encode_setup!(msg_id, wire_type);
+                Ok(writer.into_vec())
+            } |reader| {
+                let msg_id = reader.read_u64()?;
+                LairWire::ToLairX25519NewFromEntropy { msg_id }
+            },
+            ToCliX25519NewFromEntropyResponse 0x00000243 false false {
+                keystore_index: KeystoreIndex,
+                pub_key: x25519::X25519PubKey,
+            } |msg_id, wire_type| {
+                let mut writer = default_encode_setup!(msg_id, wire_type);
+                writer.write_u32(**keystore_index)?;
+                writer.write_bytes_exact(AsRef::<[u8]>::as_ref(pub_key), 32)?;
+                Ok(writer.into_vec())
+            } |reader| {
+                let msg_id = reader.read_u64()?;
+                let keystore_index = reader.read_u32()?;
+                let pub_key = reader.read_bytes(32)?;
+                let mut pub_key_array = [0; 32];
+                pub_key_array.copy_from_slice(&pub_key);
```

Comment:
Not a big deal, but if you wanted to, there's already an extension trait on this reader, you could add a fn that reads directly into an existing buffer and skip this additional copy.

Reply:
oh yeah, also i implemented `TryFrom` already and forgot about it here :sweat_smile: